src/python/pants/backend/jvm/tasks/jvm_compile/jvm_compile.py

```diff
@@ -470,6 +473,15 @@ def _record_compile_classpath(self, classpath, target, outdir):
     with open(path, 'w') as f:
       f.write(text)
 
+  def _get_directory_digest_of_dependencies(self, compile_context):
+    relative_classes_dir = fast_relpath(compile_context.classes_dir.path, get_buildroot())
+    relative_analysis_file = fast_relpath(compile_context.analysis_file, get_buildroot())
+    snapshot = self.context._scheduler.capture_snapshots((PathGlobsAndRoot(
```

Comment:
Let's batch these snapshot captures, so that we do all of the targets in one invocation, rather than doing one per loop iteration; just like in the other PR :)